cfscrape/__init__.py

```diff
@@ -272,6 +305,30 @@ def solve_challenge(self, body, domain):
 
         return result, delay
 
+    def should_mount_adapter(self, url=None):
+        if self.cloudflare_is_bypassed(url):
+            return False
+
+        if url and urlparse(url).scheme != "https":
+            return False
+
+        return True
+
+    def mount_adapter(self):
+        adapter = self.get_adapter(CaptchaProvokingCiphersRemover.url)
+
+        if not isinstance(adapter, CaptchaProvokingCiphersRemover):
+            self.mount(CaptchaProvokingCiphersRemover.url, self.captcha_adapter)
+
+    def unmount_adapter(self):
+        adapter = self.get_adapter(CaptchaProvokingCiphersRemover.url)
+
+        if isinstance(adapter, CaptchaProvokingCiphersRemover):
+            del self.adapters[CaptchaProvokingCiphersRemover.url]
```

Comment:
@lukele I'm thinking you're right about the connection pool still having open handles when the adapter is removed.
Perhaps we should call the `poolmanager`'s clear method to dispose of the adapter properly.
https://github.com/urllib3/urllib3/blob/master/src/urllib3/poolmanager.py#L169
https://github.com/urllib3/urllib3/blob/master/src/urllib3/poolmanager.py#L201-L208
```py
adapter.poolmanager.clear()
```

Reply:
To have the least impact, how about we install the adapter by default but add methods to enable/disable it. Since for each connection init_poolmanager is called if I'm not mistaken, that way we should be able to easily control whether a custom ssl context is created or not.

Reply:
4 sure, and I think it would help simplify this

Reply:
Absolutely!